src/server/helpers/collections.js

```diff
@@ -106,3 +106,32 @@ export async function getOrderedPublicCollections(from, size, entityType, orm) {
 
 	return collectionsJSON;
 }
+
+/**
+ * Fetches bbids of entities in the collection
+ * Fetches the last 'size' number of bbids with offset 'from'
+ * @param {uuid} collectionId - collectionId
+ * @param {number} from - the offset value
+ * @param {number} size - no. of last collections required
+ * @param {object} orm - the BookBrainz ORM, initialized during app setup
+ * @returns {array} - array of bbids
+ */
+export async function getCollectionItemBBIDs(collectionId, from, size, orm) {
+	const result = await orm.bookshelf.knex.raw(`
+						SELECT bookbrainz.user_collection_item.bbid
+						FROM bookbrainz.user_collection_item
+						WHERE collection_id='${collectionId}'
+						ORDER BY user_collection_item.added_at ASC
+						LIMIT ${size}
+						OFFSET ${from}`);
+	return result.rows.map(row => row.bbid);
+	// const {UserCollectionItem} = orm;
+	// const items = await new UserCollectionItem()
+	// 	.where('collection_id', collectionId)
+	// 	.orderBy('added_at')
+	// 	.fetchPage({
+	// 		debug: true
+	// 	});
+	// const itemsJSON = items.toJSON();
+	// return itemsJSON.map(item => item.bbid);
```

Comment:
`fetchPage` is not working here. `debug: true` shows two sql queries for `fetchPage` (idk why..). The problem is with second query which counts the number of items.. I think because of composite primary key

Reply:
`fetchAll` is working fine though

Reply:
I'll have a look into this. Have you checked if it behaves differently for other paginated pages?

Reply:
At first I hadn't understood that it wasn't working *at all*, as in throwing an error :D
I thought the not working part was that there were two queries instead of one.

I see the issue. It's undoubtedly related to Bookshelf not supporting composite primary keys ([#1664](https://github.com/bookshelf/bookshelf/issues/1664)).

You solved it the right way :+1:. You can remove the commented out fetchpage code I think, we'll just do it this way.